File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/SplashScreenActivity.java
package fr.frogdevelopment.pregnancycalendar;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class SplashScreenActivity extends Activity {

    // Splash screen timer
    private static final long SPLASH_TIME_OUT = 500;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_screen_activity);

        new Handler().postDelayed(() -> {
            // This method will be executed once the timer is over
            // Start your app main activity
            Intent i = new Intent(SplashScreenActivity.this, MainActivity.class);
            SplashScreenActivity.this.startActivity(i);

            // close this activity
            SplashScreenActivity.this.finish();
        }, SPLASH_TIME_OUT);
    }

}

File: app/src/main/java/fr/frogdevelopment/pregnancycalendar/SplashScreenActivity.kt
package fr.frogdevelopment.pregnancycalendar

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler

class SplashScreenActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_screen_activity)

        Handler().postDelayed({
            val i = Intent(this, MainActivity::class.java)
            startActivity(i)

            finish()
        }, SPLASH_TIME_OUT)
    }

    companion object {
        private const val SPLASH_TIME_OUT: Long = 500
    }
}